integration/tests/user_scenario/basic.rb

```diff
@@ -0,0 +1,50 @@
+require 'git_utils'
+require 'master_manipulator'
+test_name "RK-5 - C59118 - Basic User Scenario"
+
+#Init
+master_certname = on(master, puppet('config', 'print', 'certname')).stdout.rstrip
+git_environments_path = '/root/environments'
+last_commit = git_last_commit(master, git_environments_path)
+local_files_root_path = ENV['FILES'] || 'files'
+helloworld_module_path = File.join(local_files_root_path, 'modules', 'helloworld')
+
+#Manifest
+site_pp_path = File.join(git_environments_path, 'manifests', 'site.pp')
+site_pp = create_site_pp(master_certname, '  include helloworld')
+
+#Verification
+test_folder = '/tmp/manifest'
+test_file = '/tmp/manifest/random_fifth'
+notify_message_regex = /I am in the production environment/
+
+#Teardown
+teardown do
+  step "Reset Git Repo to Known Good State"
+  git_revert_environment(master, last_commit, git_environments_path)
+end
+
+#Setup
+step 'Checkout "production" Branch'
+git_on(master, 'checkout production', git_environments_path)
+
+step 'Copy "helloworld" Module to "production" Environment Git Repo'
+scp_to(master, helloworld_module_path, File.join(git_environments_path, "site", 'helloworld'))
+
+step 'Inject New "site.pp" to the "production" Environment'
+inject_site_pp(master, site_pp_path, site_pp)
+
+step 'Push Changes'
+git_add_commit_push(master, 'production', 'Update site.pp and add module.', git_environments_path)
+
+#Tests
+step 'Deploy "production" Environment via r10k'
+on(master, 'r10k deploy environment -v')
+
+agents.each do |agent|
+  step "Run Puppet Agent"
+  on(agent, puppet('agent', '--test'), :acceptable_exit_codes => 2) do |result|
```

Comment:
@zreichert The Puppet run should always produce a notify message which will always register as a change so in this case if the Puppet run didn't return "2" we know that a failure occurred. Adding "0" wouldn't necessarily hurt anything, but would technically make the validation looser.